src/plugins/intel_cpu/src/nodes/kernels/x64/random_uniform.cpp

```diff
@@ -626,10 +640,675 @@ void RandomUniform<isa>::tail(const std::vector<Vmm>& vmm_dst) {
     L(l_end);
 }
 
-template class RandomUniform<x64::avx512_core>;
-template class RandomUniform<x64::avx2>;
-template class RandomUniform<x64::sse41>;
+//////////////// MERSENNE TWISTER GENERATOR ////////////////////
+
+template <x64::cpu_isa_t isa>
+MersenneTwisterGenerator<isa>::MersenneTwisterGenerator(const MersenneTwisterGeneratorCompileParams& jcp) :
+    JitKernel(jit_name(), jcp, isa) {}
+
+template <x64::cpu_isa_t isa>
+void MersenneTwisterGenerator<isa>::generate() {
+    this->preamble();
+    registersPool = RegistersPool::create(isa, {rax, rcx, rsp, rdi, k0});
+
+    r64_dst = getReg64();
+    r64_state = getReg64();
+    r64_output_idx = getReg64();
+    r64_max_output_idx = getReg64();
+    r64_storage_capacity = getReg64();
+    r64_elements_to_generate = getReg64();
+    r64_state_accesses_count = getReg64();
+
+    mov(r64_dst,                  ptr[r64_params + GET_MERSENNE_OFFSET(dst_ptr)]);
+    mov(r64_state,                ptr[r64_params + GET_MERSENNE_OFFSET(state_ptr)]);
+    mov(r64_output_idx,           ptr[r64_params + GET_MERSENNE_OFFSET(output_idx)]);
+    mov(r64_max_output_idx,       ptr[r64_params + GET_MERSENNE_OFFSET(max_output_idx)]);
+    mov(r64_elements_to_generate, ptr[r64_params + GET_MERSENNE_OFFSET(elements_to_generate)]);
+    mov(r64_state_accesses_count, ptr[r64_params + GET_MERSENNE_OFFSET(state_accesses_count)]);
+    INIT_1_ELEM_T_VAL(storage_capacity, static_cast<uint64_t>(vlen / sizeof(uint32_t)), r64_storage_capacity, uint64_t);
```

Comment:
It's better to move constant directly to the register `mov(r64_storage_capacity, static_cast<uint64_t>(vlen / sizeof(uint32_t)))`

Reply:
Applied!